swatch-core/src/main/java/org/candlepin/subscriptions/db/model/BillingProvider.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.db.model;
+
+import java.util.Map;
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+import org.candlepin.subscriptions.utilization.api.model.BillingProviderType;
+
+public enum BillingProvider implements StringValueEnum<BillingProviderType> {
+  EMPTY("", BillingProviderType.EMPTY),
+  RED_HAT("Red hat", BillingProviderType.RED_HAT),
+  AWS("Aws", BillingProviderType.AWS),
+  GCP("Gcp", BillingProviderType.GCP),
+  AZURE("Azure", BillingProviderType.AZURE),
+  ORACLE("Oracle", BillingProviderType.ORACLE),
+  _ANY("_ANY", BillingProviderType._ANY); // NOSONAR
+
+  private static final Map<String, BillingProvider> VALUE_ENUM_MAP =
+      StringValueEnum.initializeImmutableMap(BillingProvider.class);
+
+  private final String value;
+  private final BillingProviderType openApiEnum;
+
+  BillingProvider(String value, BillingProviderType openApiEnum) {
+    this.value = value;
+    this.openApiEnum = openApiEnum;
+  }
+
+  /**
+   * Parse the BillingProvider from its string representation
+   *
+   * @param value String representation of the BillingProvider, as seen in a host record
+   * @return the BillingProvider enum
+   */
+  public static BillingProvider fromString(String value) {
+    return StringValueEnum.getValueOf(BillingProvider.class, VALUE_ENUM_MAP, value, EMPTY);
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  @Override
+  public BillingProviderType asOpenApiEnum() {
+    return openApiEnum;
+  }
+
+  /** JPA converter for BillingProvider */
+  @Converter(autoApply = true)
+  public static class EnumConverter implements AttributeConverter<BillingProvider, String> {
```

Comment:
Wondering if there is something we can do to make this a little more DRY.  This will be the third time that we're declaring an enum that implements `StringValueEnum`.

swatch-core/src/main/java/org/candlepin/subscriptions/db/model/Usage.java
swatch-core/src/main/java/org/candlepin/subscriptions/db/model/ServiceLevel.java

cc: @kahowell

Reply:
That's a good thought. Happy to either timebox it to an hour or create a tech-debt card for the backlog for this. We already worked on getting this DRY, (That's how StringValueEnum came about in the first place); if there are improvements to be had, it's probably in pulling more things into that interface.